src/Microsoft.Diagnostics.Monitoring.Options/S3StorageEgressProviderOptions.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.WebApi;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.Egress.S3
+{
+    /// <summary>
+    /// Egress provider options for file system egress.
+    /// </summary>
+    internal sealed class S3StorageEgressProviderOptions :
```

Comment:
I would like to see unit tests for the custom validation portions. This example might be a bit overboard for these options, but here is how we are testing other types of options classes that contain validation attributes and implement IValidatableObject: https://github.com/dotnet/dotnet-monitor/blob/main/src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/CollectionRuleOptionsTests.cs